pkg/server/download.go

```diff
@@ -293,9 +293,11 @@ func (s *server) downloadFileFromPresignedUrl(ctx context.Context, url string, t
 		return 0, err
 	}
 
-	err = os.Rename(tempPath, targetLocation)
+	err = fileSystemSafeRename(tempPath, targetLocation)
+
+	// Catch the error, read in the data from tempPath and write to targetLocation
```

Comment:
the comment here is a little confusing, isn't the error being caught after the data is read and written from tempPath to targetLocation?